starter_code/app.js

```diff
@@ -2,27 +2,97 @@ require('dotenv').config()
 
 const express = require('express');
 const hbs = require('hbs');
-
-// require spotify-web-api-node package here:
-
-
-
+const SpotifyWebApi = require("spotify-web-api-node");
 const app = express();
 
+
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
-
-
-// setting the spotify-api goes here:
-
-
+// hbs.registerPartials(__dirname + '/views/partials');
 
 
 
+// setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+});
+
+// Retrieve an access token
+spotifyApi
+    .clientCredentialsGrant()
+    .then(data => {
+        spotifyApi.setAccessToken(data.body["access_token"]);
+    })
+    .catch(error => {
+        console.log("Something went wrong when retrieving an access token", error);
+    });
 
 // the routes go here:
-
-
-
-app.listen(3000, () => console.log("My Spotify project running on port 3000 🎧 🥁 🎸 🔊"));
+app.get('/', (req, res) => {
+    res.render('index')
+})
+
+app.get('/artists', (req, res) => {
+    // res.send(req.query.artist)
+    let searchedArtist = req.query.artist
+
+    spotifyApi
+        .searchArtists(searchedArtist)
+        .then(data => {
+            // console.log("The received data from the API: ", data.body.artists.items);
+            let artistsArr = data.body.artists.items;
```

Comment:
these two lines you can combine into one using the ES6 syntax directly on line 45.